### PR TITLE
fix: enable static link for windows-msvc

### PR DESCRIPTION
**Pull request by Zagrios**
The generated `.node` file for Windows require certain DLLs which may not be present, especially in applications used by end-users which can lead to a `Missing DLL error.

It was my case with some users with my application so I've switch to `sharp` for now.

With [Dependencies](https://github.com/lucasg/Dependencies) we can see the amount of needed DLLs is greatly reduce with static link :
|before|after|
|--|--|
|image|image|

The size of the generated `.node` file is a slightly larger tho
|before|after|
|--|--|
|4.12mo|4.29mo|

### CODE REVIEW

**Comment by yisibl** [comment]
Thanks!

There is a test case that needs to be skipped, so please rebase the main branch.

File: .cargo/config.toml [hunk 1]
```diff
@@ -10,3 +10,6 @@ rustflags = [
 [target.aarch64-unknown-linux-musl]
 linker = "aarch64-linux-musl-gcc"
 rustflags = ["-C", "target-feature=-crt-static"]
+
+[target.x86_64-pc-windows-msvc]
+rustflags = ["-C", "target-feature=+crt-static"]
```
Comment (yisibl) [inline .cargo/config.toml:15]: Please add a line break at the end.

**Comment by Zagrios** [comment]
@yisibl done 😉

**Comment by yisibl** [comment]
I would like to confirm if the users who are prompted for missing `dll` are doing so because they don't have "[Visual C++ Redistributable](https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170)" installed.

**Comment by Zagrios** [comment]
Yeah exactly

**Comment by yisibl** [comment]
It looks like x86 needs to be added as well:

```yml
[target.i686-pc-windows-msvc]
rustflags = ["-C", "target-feature=+crt-static"]
```

**Comment by yisibl** [comment]
Currently `x86_64-unknown-linux-musl` is reporting errors in the CI, and I've been troubleshooting it for two days with no luck.

**Comment by Zagrios** [comment]
> Currently `x86_64-unknown-linux-musl` is reporting errors in the CI, and I've been troubleshooting it for two days with no luck.

Yeah I saw your PR 😅

**Comment by yisibl** [comment]
By the way, what app are you using resvg-js in?

**Comment by Zagrios** [comment]
@yisibl yup, it's working perfectly fine now 👌

**Pull request merged**